### ISSUE TITLE
False positive function-redefined for methods with the typing.overload decorator

### ISSUE THREAD

**Issue by brycepg** [PyCQA/pylint#2239]
`function-redefined` should ignore functions/methods with the `typing.overload` decorator since these function stubs are only used to give type information.

[Example](https://docs.python.org/3/library/typing.html#typing.overload):

```
from typing import overload

@overload # Ignore
def process(response: None) -> None:
    ...
@overload # Ignore
def process(response: int) -> Tuple[int, str]:
    ...
@overload # Ignore
def process(response: bytes) -> str:
    ...
def process(response):
    <actual implementation>
```

**Comment by PCManticore** [comment]
@SLAPaper Feel free to disable it for now, we didn't get to work on it just yet.

**Comment by SLAPaper** [comment]
@PCManticore Thanks!

**Comment by AWhetter** [comment]
Closing as a duplicate of #1581
